lib/query.js

```diff
@@ -3783,24 +3784,35 @@ Query.prototype.http = function(args, options, internalOptions) {
     return new Promise(function(resolve, reject) {
       var options = {
         url: url,
-        headers: {
-          "Accept": "*/*" ,
-          "Accept-Encoding": "deflate;q=1, gzip;q=0.5" ,
-          "Host": "httpbin.org" ,
-          "User-Agent": "RethinkDB/2.0.2"
-        }
+        encoding: null,
+        timeout: internalOptions.timeout || 30000,
+        maxRedirects: internalOptions.redirects || 1,
+        qs: internalOptions.params,
+        headers: httpUtil.headers(internalOptions.header)
       };
       request.get(options, function(err, httpResponse, body) {
         if (err) {
           reject(err);
         }
         else {
-          //TODO Handle more options
-          var response = body;
-          if (httpResponse.headers['content-type'] === 'application/json') {
-            response = JSON.parse(response);
+          var response = body.toString('utf8');
+          var format = httpUtil.responseFormat(internalOptions.resultFormat, httpResponse.headers['content-type']);
```

Comment:
Same thing, you want `options.resultFormat` here. `internalOptions` are options we build in reqlite that currently only track non atomic query (for writes).